sdk/identity/azure_identity/src/credentials/cache.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use async_lock::RwLock;
+use azure_core::credentials::AccessToken;
+use futures::Future;
+use std::collections::HashMap;
+use tracing::trace;
+
+#[derive(Debug)]
+pub(crate) struct TokenCache(RwLock<HashMap<Vec<String>, AccessToken>>);
+
+impl TokenCache {
+    pub(crate) fn new() -> Self {
```

Comment:
Tip: the type is already `pub(crate)` (and doesn't need to be; see previous comment). No need to declare these as `pub(crate)` when `pub` will suffice.